### ISSUE TITLE
json转化问题

### ISSUE THREAD

**Issue by youziXJJ** [wavesplatform/WavesJ#25]
The problem I found is that the TransferTransactionV2 object generated by the signature will have the parent class attribute. When converting it to json and converting json to TransferTransactionV2 object, it cannot be converted. This is the source of the problem.

**Comment by KardanovIR** [comment]
You're trying to 
1) Create a transaction
2) Sign it
3) Serialize to JSON
4) Send it to another person
5) Deserialize from the JSON
6) Add another signature (proof).

Am I right?

**Comment by youziXJJ** [comment]
yes,but 5 error

**Comment by youziXJJ** [comment]
this is my json:
{"data":{"senderPublicKey":"Ds9m7XEAq6nqPy5jn6ZX5hUVgTfTd3vGSoWTXADBdojh","amount":100000000,"attachment":"EkznZwTjscgi2DZmnGvxwX","fee":100000,"recipient":"3PJfcKzDdr4r3V9nE2QmJ51c4jrXSpbdDWe","type":4,"timestamp":1550635480751},"id":"6QG6mkQLBVnQes41CyUeBqWrSs6CyrBHWQh1TefHM3jn","json":"{\"senderPublicKey\":\"Ds9m7XEAq6nqPy5jn6ZX5hUVgTfTd3vGSoWTXADBdojh\",\"amount\":100000000,\"attachment\":\"EkznZwTjscgi2DZmnGvxwX\",\"signature\":\"5asraKZuPFxhVHNwA9or1yA5zsNhfxWpjgf3KRqV1Q6gnCatPYJiLkbkpHggivKdHRRXurjdkJdCcpd6nzWxvfEk\",\"proofs\":[\"5asraKZuPFxhVHNwA9or1yA5zsNhfxWpjgf3KRqV1Q6gnCatPYJiLkbkpHggivKdHRRXurjdkJdCcpd6nzWxvfEk\"],\"fee\":100000,\"recipient\":\"3PJfcKzDdr4r3V9nE2QmJ51c4jrXSpbdDWe\",\"id\":\"6QG6mkQLBVnQes41CyUeBqWrSs6CyrBHWQh1TefHM3jn\",\"type\":4,\"timestamp\":1550635480751}","proofs":["5asraKZuPFxhVHNwA9or1yA5zsNhfxWpjgf3KRqV1Q6gnCatPYJiLkbkpHggivKdHRRXurjdkJdCcpd6nzWxvfEk"]}

**Comment by youziXJJ** [comment]
this is error info:
{
  "error": 1,
  "message": "failed to parse json message",
  "cause": null,
  "validationErrors": {
    "obj": [
      {
        "msg": [
          "'type' is undefined on object: {\"data\":{\"senderPublicKey\":\"Ds9m7XEAq6nqPy5jn6ZX5hUVgTfTd3vGSoWTXADBdojh\",\"amount\":100000000,\"attachment\":\"EkznZwTjscgi2DZmnGvxwX\",\"fee\":100000,\"recipient\":\"3PJfcKzDdr4r3V9nE2QmJ51c4jrXSpbdDWe\",\"type\":4,\"timestamp\":1550635480751},\"id\":\"6QG6mkQLBVnQes41CyUeBqWrSs6CyrBHWQh1TefHM3jn\",\"json\":\"{\\\"senderPublicKey\\\":\\\"Ds9m7XEAq6nqPy5jn6ZX5hUVgTfTd3vGSoWTXADBdojh\\\",\\\"amount\\\":100000000,\\\"attachment\\\":\\\"EkznZwTjscgi2DZmnGvxwX\\\",\\\"signature\\\":\\\"5asraKZuPFxhVHNwA9or1yA5zsNhfxWpjgf3KRqV1Q6gnCatPYJiLkbkpHggivKdHRRXurjdkJdCcpd6nzWxvfEk\\\",\\\"proofs\\\":[\\\"5asraKZuPFxhVHNwA9or1yA5zsNhfxWpjgf3KRqV1Q6gnCatPYJiLkbkpHggivKdHRRXurjdkJdCcpd6nzWxvfEk\\\"],\\\"fee\\\":100000,\\\"recipient\\\":\\\"3PJfcKzDdr4r3V9nE2QmJ51c4jrXSpbdDWe\\\",\\\"id\\\":\\\"6QG6mkQLBVnQes41CyUeBqWrSs6CyrBHWQh1TefHM3jn\\\",\\\"type\\\":4,\\\"timestamp\\\":1550635480751}\",\"proofs\":[\"5asraKZuPFxhVHNwA9or1yA5zsNhfxWpjgf3KRqV1Q6gnCatPYJiLkbkpHggivKdHRRXurjdkJdCcpd6nzWxvfEk\"]}"
        ],
        "args": []
      }
    ]
  }
}

**Comment by br0x** [comment]
"json":"{"senderPublicKey": ...
is just invalid json syntax

**Comment by DrBlast** [comment]
You can send transfer tx in several ways:
1. Create transfer transaction and send it:
```
TransferTransactionV2 transferWavesTx = Transactions.makeTransferTx(acc, "<recipient address>", <amount>, Asset.WAVES, FEE, Asset.WAVES, "some attachment");
node.send(transferWavesTx);
```
2. If you have a json, it should be deserialized usinf  WavesJsonMapper:
```
WavesJsonMapper wavesJsonMapper = new WavesJsonMapper(chainId);
Transaction deserialized = mapper.readValue(json, TransferTransactionV2.class);
node.send(deserialized);
```
valid transfer tx json is:
```
{
  "type" : 4,
  "id" : "sLsqf4rf6mgakFo5X24pK7psN1ZZTQy7xwYBzHWVqKZ",
  "sender" : "3P44vnPZv4DfTGoJfhahZPRZUUQYgSE32cu",
  "senderPublicKey" : "BVvbdjve9mwmgzM4ZwuoVPvXpSTpPiWhSwEkBBDUS465",
  "fee" : 100000,
  "timestamp" : 1550656696171,
  "proofs" : [ "d2Hiyxpy5hmdmcm2ExGaSYDkLYq8pC6EFa8nmKvNUqFdssNKxtLLokfFzgsR1fhaYPuqkhAxhbTfT1RPM2cCQZ1" ],
  "version" : 2,
  "recipient" : "3PKCxrT4FPWnYTAPHPtrZ3oMz1pQf3LhejC",
  "assetId" : "6UfBupFwFnWTrnFDWiR73LQjPxRY38CgZ6Bu4m4SySQY",
  "feeAssetId" : null,
  "feeAsset" : null,
  "amount" : 1000,
  "attachment" : "",
  "height" : 1404709
}
```

3. You can also create a Trasfet tx object and sign it separatly
```
TransferTransactionV2 unsignedTx = new TransferTransactionV2(voteBankPublicKey, recipient, <amount>, assetId,
               <fee>, Asset.WAVES, new ByteString("some attachment"), System.currentTimeMillis(), new ArrayList<>());
        String signature = acc.sign(tx);
        signedTx = unsignedTx.withProof(0, new ByteString(signature));
        String txId = node.send(signedTx);
```
4. And the simplest way is use method 
`node.transfer(...)`

**Comment by youziXJJ** [comment]
This is not the case, I need to transfer the signed transaction to someone else.

**Comment by DrBlast** [comment]
I described 4 ways how you can transfer waves to recipient using wavesJ.
What problem do you have?

**Comment by youziXJJ** [comment]
Ok, I solved this problem with V0.7, thank you for your support.
